tests/qat/test_quantum_backend.py

```diff
@@ -677,3 +681,37 @@ def test_empty_channels_removed(self):
                     for p in positions
                 ]
             )
+
+    def test_create_duration_timeline_mapping(self):
+        """Test that instructions are correctly mapped to expected channels."""
+        hw = get_test_model()
+        qubit = hw.get_qubit(0)
+
+        builder = (
+            get_builder(hw).X(qubit, np.pi / 2.0).measure_mean_z(qubit).synchronize(qubit)
+        )
```

Comment:
can you parametrise the builder ? I'd ideally play with different combinations of instructions and see what happens

Reply:
We could, however that would also affect the assertions. Potentially better to add separate tests in the instances you'd like to test.